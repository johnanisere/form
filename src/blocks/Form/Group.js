import styled from "styled-components";
import Section from "../../elements/Section";

const Group = styled(Section)`
  flex-direction: column;
  padding: 20px;
`;

export default Group;
