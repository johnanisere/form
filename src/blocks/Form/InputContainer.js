import styled, { css } from "styled-components";
import Article from "../../elements/Article";

const InputContainer = styled(Article)`
  ${({ special }) =>
    special &&
    css`
      display: flex;
      justify-content: center;
      padding: 20px;
      input {
        font-size: 14px;
        color: rgb(255, 255, 255);
        outline: none;
        background: linear-gradient(rgb(99, 113, 199), rgb(85, 99, 193));
        &:hover ,&:focus{
          box-shadow: 2px 4px 11px 1px rgba(0,0,0,.2)
          transform: translateY(-1);
        }
      }
    `}
`;

export default InputContainer;
