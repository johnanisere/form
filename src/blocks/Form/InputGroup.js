import styled, { css } from "styled-components";
import P from "../../elements/P";

const InputGroup = styled(P)`
  ${({ special }) =>
    special &&
    css`
      display: flex;
      justify-content: center;
      padding: 20px;
      input {
        font-size: 14px;
        color: rgb(255, 255, 255);
        outline: none;
        background: linear-gradient(rgb(99, 113, 199), rgb(85, 99, 193));
        cursor: pointer;
      }
    `}
`;

export default InputGroup;
