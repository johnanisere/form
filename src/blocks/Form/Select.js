import styled from "styled-components";
import MySelect from "../../elements/Select";

const Select = styled(MySelect)`
  width: 100%;
  display: block;
  box-sizing: border-box;
  border: 1px solid #e3e5e7;
  height: 120px;
  font-size: 14px;
  line-height: 1.42857143;
  padding: 18px;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
`;

export default Select;
