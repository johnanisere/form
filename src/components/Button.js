import React from "react";

import Form from "../blocks/Form";

const Button = ({ fetching }) => {
  const value = fetching ? "Processing..." : "Submit";
  return (
    <Form.InputGroup special={true}>
      <Form.Input type="submit" disabled={fetching} value={value} />
    </Form.InputGroup>
  );
};

export default Button;
