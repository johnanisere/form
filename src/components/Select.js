import React from "react";

import Form from "../blocks/Form";

const Select = ({
  id,
  name,
  value,
  onChange,
  multiple,
  selected,
  required,
  list = []
}) => (
  <Form.Select
    id={id}
    name={name}
    value={value}
    onChange={onChange}
    multiple={multiple}
    required={required}
  >
    {multiple || <option value="">Select</option>}
    {list.map((value, key) => (
      <option value={value} key={key} selected={selected}>
        {value}
      </option>
    ))}
  </Form.Select>
);

export default Select;
