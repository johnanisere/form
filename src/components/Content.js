import React from "react";

import Form from "../blocks/Form";
import InputGroup from "./InputGroup";
import { getTimeFrame, onStripByKey } from "../helpers/utils";

const Content = ({
  city,
  title,
  skills,
  salary,
  allSkills,
  description,
  backgrounds,
  allBackgrounds,
  onHandleChange,
  latest_end_date,
  required = true,
  selection_process,
  earliest_start_date,
  applications_close_date
}) => (
  <Form.Group>
    <InputGroup
      id="title"
      name="title"
      type="text"
      title="Title"
      value={title}
      maxLength={100}
      required={required}
      onChange={onHandleChange}
    />
    <InputGroup
      id="skills"
      name="skills"
      type="select"
      title="Skills"
      value={skills}
      multiple={true}
      required={required}
      onChange={onHandleChange}
      list={onStripByKey(allSkills)}
    />
    <InputGroup
      id="salary"
      name="salary"
      type="number"
      title="Salary"
      value={salary}
      required={required}
      onChange={onHandleChange}
    />
    <InputGroup
      type="text"
      id="description"
      name="description"
      title="Description"
      value={description}
      required={required}
      onChange={onHandleChange}
    />
    <InputGroup
      multiple={true}
      type="select"
      id="backgrounds"
      name="backgrounds"
      title="Backgrounds"
      value={backgrounds}
      required={required}
      onChange={onHandleChange}
      list={onStripByKey(allBackgrounds)}
    />
    <InputGroup
      type="text"
      required={required}
      id="selection_process"
      name="selection_process"
      title="Selection process"
      onChange={onHandleChange}
      value={selection_process}
    />
    <InputGroup
      id="city"
      name="city"
      type="location"
      title="City"
      value={city}
      required={required}
      onChange={onHandleChange}
    />
    <InputGroup
      required={required}
      id="latest_end_date"
      type="datetime-local"
      name="latest_end_date"
      title="Latest end date"
      value={latest_end_date}
      onChange={onHandleChange}
    />
    <InputGroup
      required={required}
      type="datetime-local"
      id="earliest_start_date"
      onChange={onHandleChange}
      name="earliest_start_date"
      title="Earliest start date"
      value={earliest_start_date}
    />
    <InputGroup
      required={required}
      type="datetime-local"
      onChange={onHandleChange}
      id="applications_close_date"
      min={getTimeFrame(30, 90).min}
      max={getTimeFrame(30, 90).max}
      name="applications_close_date"
      title="Applications close date"
      value={applications_close_date}
    />
  </Form.Group>
);

Content.defaultProps = {
  title: "qwertyui",
  skills: [""],
  salary: "",
  required: true,
  city: "albenia",
  description: "",
  backgrounds: [""],
  onChange: () => {},
  latest_end_date: "2018-06-12T19:30",
  selection_process: "",
  earliest_start_date: "2018-06-12T19:30",
  applications_close_date: new Date().toISOString().slice(0, -1)
};
export default Content;
