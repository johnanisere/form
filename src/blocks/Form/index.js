import styled from "styled-components";

import Input from "./Input";
import Group from "./Group";
import Title from "./Title";
import Select from "./Select";
import InputGroup from "./InputGroup";
import InputLabel from "./InputLabel";
import MyForm from "../../elements/Form";
import InputContainer from "./InputContainer";

const Form = styled(MyForm)`
  width: 100%;
  max-width: 580px;
  box-shadow: rgba(63, 63, 68, 0.05) 0px 0px 0px 1px,
    rgba(63, 63, 68, 0.15) 0px 1px 3px 0px;
  border-radius: 3px;
  margin: 50px 0;

  .geosuggest__input {
    width: 100%;
    display: block;
    box-sizing: border-box;
    padding: 4px 18px;
    border: 1px solid #e3e5e7;
    height: 40px;
    font-size: 14px;
    line-height: 1.42857143;
    color: #555;
    background-color: #fff;
    background-image: none;
    border: 1px solid #ccc;
    border-radius: 4px;
    box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
    transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
    &:focus {
      border-width: 2px !important;
      border-style: solid !important;
      border-color: rgb(92, 106, 196) !important;
      border-image: initial !important;
      outline: white;
    }
  }
  .geosuggest__suggests {
    list-style: none;
    padding: 0;
    background: #ffff;
    margin: 0;
    box-shadow: rgba(63, 63, 68, 0.05) 0px 0px 0px 1px,
      rgba(63, 63, 68, 0.15) 0px 1px 3px 0px;
    border-radius: 3px;
    li {
      padding: 0.5em 20px;
      cursor: pointer;
      &:hover {
        background: #f5f5f5;
      }
    }
  }
  .geosuggest__suggests--hidden {
    display: none;
  }
`;

Form.Title = Title;
Form.Input = Input;
Form.Group = Group;
Form.Select = Select;
Form.Label = InputLabel;
Form.InputGroup = InputGroup;
Form.InputContainer = InputContainer;

export default Form;
