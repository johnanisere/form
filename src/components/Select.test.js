import React from "react";
import { shallow } from "enzyme";

import Select from "./Select";
import Button from "./Button";
import Content from "./Content";
import Form from "../blocks/Form";

const props = {
  id: "",
  name: "",
  value: "",
  onChange: () => {},
  multiple: false,
  required: true,
  list: []
};

it("Select component exists", () => {
  const wrapper = shallow(<Select {...props} />);
  expect(wrapper.exists()).toBe(true);
});

it("Select component renders", () => {
  const wrapper = shallow(<Select {...props} />);
  expect(wrapper).toMatchSnapshot();
});

it(`Select should contain "Select"`, () => {
  const wrapper = shallow(<Select {...props} />);

  expect(wrapper.find(Form.Select)).toHaveLength(1);
});
