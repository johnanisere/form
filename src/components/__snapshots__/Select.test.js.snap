// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select component renders 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Select
    id=""
    list={Array []}
    multiple={false}
    name=""
    onChange={[Function]}
    required={true}
    value=""
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <option
          value=""
        >
          Select
        </option>,
        Array [],
      ],
      "id": "",
      "multiple": false,
      "name": "",
      "onChange": [Function],
      "required": true,
      "value": "",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Select",
          "value": "",
        },
        "ref": null,
        "rendered": "Select",
        "type": "option",
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-gzVnrw",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  display: block;
  box-sizing: border-box;
  border: 1px solid #e3e5e7;
  height: 120px;
  font-size: 14px;
  line-height: 1.42857143;
  padding: 18px;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
",
        ],
      },
      "displayName": "Styled(styled.select)",
      "foldedComponentIds": Array [
        "sc-bZQynM",
      ],
      "render": [Function],
      "styledComponentId": "sc-gzVnrw",
      "target": "select",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <option
            value=""
          >
            Select
          </option>,
          Array [],
        ],
        "id": "",
        "multiple": false,
        "name": "",
        "onChange": [Function],
        "required": true,
        "value": "",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Select",
            "value": "",
          },
          "ref": null,
          "rendered": "Select",
          "type": "option",
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-gzVnrw",
          "isStatic": true,
          "rules": Array [
            "
  width: 100%;
  display: block;
  box-sizing: border-box;
  border: 1px solid #e3e5e7;
  height: 120px;
  font-size: 14px;
  line-height: 1.42857143;
  padding: 18px;
  color: #555;
  background-color: #fff;
  background-image: none;
  border: 1px solid #ccc;
  border-radius: 4px;
  box-shadow: inset 0 1px 1px rgba(0, 0, 0, 0.075);
  transition: border-color ease-in-out 0.15s, box-shadow ease-in-out 0.15s;
  &:focus {
    border-width: 2px !important;
    border-style: solid !important;
    border-color: rgb(92, 106, 196) !important;
    border-image: initial !important;
    outline: white;
  }
",
          ],
        },
        "displayName": "Styled(styled.select)",
        "foldedComponentIds": Array [
          "sc-bZQynM",
        ],
        "render": [Function],
        "styledComponentId": "sc-gzVnrw",
        "target": "select",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
